include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -1754,6 +1973,247 @@ namespace internal
           }
       }
 
+
+
+      /**
+       * This evaluates the mapping information on a range of cells calling
+       * into the tensor product interpolators of the matrix-free framework,
+       * using a polynomial expansion of the cell geometry in terms of
+       * MappingQ.
+       */
+      template <int dim, typename Number, typename VectorizedArrayType>
+      void
+      compute_range_mapping_q(
+        const unsigned int begin_face,
+        const unsigned int end_face,
+        const std::vector<
+          FaceToCellTopology<VectorizedArrayType::n_array_elements>> &faces,
+        const std::vector<GeometryType> &                             face_type,
+        const std::vector<bool> &    process_face,
+        const UpdateFlags            update_flags_faces,
+        const AlignedVector<double> &plain_quadrature_points,
+        const FE_DGQ<dim> &          fe_geometry,
+        MappingInfoStorage<dim - 1, dim, Number, VectorizedArrayType> &my_data)
+      {
+        // we want to use vectorization for computing the quantities, but must
+        // evaluate the geometry in double precision; thus, for floats we need
+        // to do things in two sweeps and convert the final result
+        constexpr unsigned int n_simd = VectorizedArrayType::n_array_elements;
+        constexpr bool         use_float = std::is_same<Number, float>::value;
+        using VectorizedDouble =
+          VectorizedArray<double,
+                          ((use_float && n_simd > 1) ?
+                             VectorizedArrayType::n_array_elements / 2 :
+                             VectorizedArrayType::n_array_elements)>;
+        constexpr unsigned int n_simd_d = VectorizedDouble::n_array_elements;
+
+        ShapeInfo<VectorizedDouble> shape_info;
+        shape_info.reinit(my_data.descriptor[0].quadrature_1d, fe_geometry);
+
+        const unsigned int n_q_points = my_data.descriptor[0].n_q_points;
+        const unsigned int n_mapping_points =
+          shape_info.dofs_per_component_on_cell;
```

Comment:
I am wondering if this not a limitation. We might want to have mapping `mapping_degree < fe_degree` particularly in the case of high-order FEM.